pkg/tsdb/elasticsearch/data_query.go

```diff
@@ -56,21 +57,29 @@ func (e *elasticsearchDataQuery) execute() (*backend.QueryDataResponse, error) {
 		if err := e.processQuery(q, ms, from, to); err != nil {
 			mq, _ := json.Marshal(q)
 			e.logger.Error("Failed to process query to multisearch request builder", "error", err, "query", string(mq), "queriesLength", len(queries), "duration", time.Since(start), "stage", es.StagePrepareRequest)
-			return &backend.QueryDataResponse{}, err
+			return createPluginErrorResponse(e.dataQueries[0].RefID, err), err
 		}
 	}
 
 	req, err := ms.Build()
 	if err != nil {
 		mqs, _ := json.Marshal(e.dataQueries)
 		e.logger.Error("Failed to build multisearch request", "error", err, "queriesLength", len(queries), "queries", string(mqs), "duration", time.Since(start), "stage", es.StagePrepareRequest)
-		return &backend.QueryDataResponse{}, err
+		return createPluginErrorResponse(e.dataQueries[0].RefID, err), err
 	}
 
 	e.logger.Info("Prepared request", "queriesLength", len(queries), "duration", time.Since(start), "stage", es.StagePrepareRequest)
 	res, err := e.client.ExecuteMultisearch(req)
 	if err != nil {
-		return &backend.QueryDataResponse{}, err
+		queryRes := &backend.QueryDataResponse{}
+		if res.Status < 500 {
+			// We don't want to show request errors in the UI and therefore we change it to a generic error
+			// We are keeping the original error in the logs
+			queryRes = createPluginErrorResponse(e.dataQueries[0].RefID, errors.New("internal server error"))
+		} else {
+			queryRes = createDownstreamErrorResponse(e.dataQueries[0].RefID, errors.New("internal server error"))
+		}
+		return queryRes, err
```

Comment:
That's unreachable code now. Thus we are never passing the original `err` down.

Reply:
Yeah, you are right that it was unreachable. I tried to fix it through https://github.com/grafana/grafana/pull/76031/commits/51ecde80e22d20d2bdd77de7b9a4baeeeed05aea. Didn't want to do too big of changes, but curious of what do you think.